persist/storer.go

```diff
@@ -79,14 +82,17 @@ func NewFileStore(storePath string, l log.Logger) (Storer, error) {
 	// Create the external directory for user-generated json
 	storeDir := filepath.Dir(store.path)
 	if _, err := os.Stat(storeDir); err != nil {
-		if err = os.MkdirAll(storeDir, 0755); err != nil {
+		if err = os.MkdirAll(storeDir, dirFilePerm); err != nil {
 			return nil, fmt.Errorf("store directory in %s could not be created", storeDir)
 		}
 	}
 
 	stat, err := os.Stat(store.path)
 	// Store file doesn't exist yet
 	if err != nil {
+		if _, err = os.OpenFile(store.path, os.O_CREATE|os.O_WRONLY, filePerm); err != nil {
+			return nil, errors.Errorf("store directory not writable: %s", storeDir)
```

Comment:
Don't think this is ok. Other error could happen here, not only not writable. You must check if the eror is that running `os.IsPermission(err)`